generate/testdata/snapshots/TestGenerateWithConfig-OptionalPointerOmitEmpty-testdata-queries-generated.go

```diff
@@ -0,0 +1,875 @@
+		graphql.NoUnmarshalJSON
+	}
+	firstPass.__OmitEmptyQueryInput = v
+
+	err := json.Unmarshal(b, &firstPass)
+	if err != nil {
+		return err
+	}
+
+	{
+		dst := &v.Dt
+		src := firstPass.Dt
+		if len(src) != 0 && string(src) != "null" {
+			*dst = new(time.Time)
+			err = testutil.UnmarshalDate(
+				src, *dst)
+			if err != nil {
+				return fmt.Errorf(
+					"unable to unmarshal __OmitEmptyQueryInput.Dt: %w", err)
+			}
+		}
+	}
+	return nil
+}
+
+type __premarshal__OmitEmptyQueryInput struct {
+	Query *UserQueryInput `json:"query,omitempty"`
+
+	Queries []*UserQueryInput `json:"queries,omitempty"`
+
+	Dt json.RawMessage `json:"dt,omitempty"`
+
+	Tz *string `json:"tz,omitempty"`
+
+	TzNoOmitEmpty *string `json:"tzNoOmitEmpty"`
+}
+
+func (v *__OmitEmptyQueryInput) MarshalJSON() ([]byte, error) {
+	premarshaled, err := v.__premarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(premarshaled)
+}
+
+func (v *__OmitEmptyQueryInput) __premarshalJSON() (*__premarshal__OmitEmptyQueryInput, error) {
+	var retval __premarshal__OmitEmptyQueryInput
+
+	retval.Query = v.Query
+	retval.Queries = v.Queries
+	{
+
+		dst := &retval.Dt
+		src := v.Dt
+		if src != nil {
+			var err error
+			*dst, err = testutil.MarshalDate(
+				src)
+			if err != nil {
+				return nil, fmt.Errorf(
+					"unable to marshal __OmitEmptyQueryInput.Dt: %w", err)
+			}
+		}
+	}
+	retval.Tz = v.Tz
+	retval.TzNoOmitEmpty = v.TzNoOmitEmpty
+	return &retval, nil
+}
+
+// __PointersQueryInput is used internally by genqlient
+type __PointersQueryInput struct {
+	Query *UserQueryInput `json:"query,omitempty"`
+	Dt    time.Time       `json:"-"`
+	Tz    *string         `json:"tz,omitempty"`
+}
+
+// GetQuery returns __PointersQueryInput.Query, and is useful for accessing the field via an interface.
+func (v *__PointersQueryInput) GetQuery() *UserQueryInput { return v.Query }
+
+// GetDt returns __PointersQueryInput.Dt, and is useful for accessing the field via an interface.
+func (v *__PointersQueryInput) GetDt() time.Time { return v.Dt }
+
+// GetTz returns __PointersQueryInput.Tz, and is useful for accessing the field via an interface.
+func (v *__PointersQueryInput) GetTz() *string { return v.Tz }
+
+func (v *__PointersQueryInput) UnmarshalJSON(b []byte) error {
+
+	if string(b) == "null" {
+		return nil
+	}
+
+	var firstPass struct {
+		*__PointersQueryInput
+		Dt json.RawMessage `json:"dt"`
+		graphql.NoUnmarshalJSON
+	}
+	firstPass.__PointersQueryInput = v
+
+	err := json.Unmarshal(b, &firstPass)
+	if err != nil {
+		return err
+	}
+
+	{
+		dst := &v.Dt
+		src := firstPass.Dt
+		if len(src) != 0 && string(src) != "null" {
+			err = testutil.UnmarshalDate(
+				src, dst)
+			if err != nil {
+				return fmt.Errorf(
+					"unable to unmarshal __PointersQueryInput.Dt: %w", err)
+			}
+		}
+	}
+	return nil
+}
+
+type __premarshal__PointersQueryInput struct {
+	Query *UserQueryInput `json:"query,omitempty"`
+
+	Dt json.RawMessage `json:"dt"`
```

Comment:
here's an example, except this example doesn't matter since the `RawMessage` will never actually be null, but if you put the same config on a different field it would.